paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -196,12 +199,35 @@ def create(self, kube_client: KubeClient) -> None:
         self.ensure_pod_disruption_budget(kube_client)
         self.sync_horizontal_pod_autoscaler(kube_client)
 
+    def deep_delete_and_create(self, kube_client: KubeClient) -> None:
+        self.logging.debug(
+            f"Deep Delete {self.kube_deployment.service} {self.kube_deployment.instance}."
+        )
+        self.deep_delete(kube_client)
+        while self.kube_deployment in set(list_all_deployments(kube_client)):
```

Comment:
Unfortunately, sometimes some pods are stuck in terminating and cannot be deleted. So if the deployment cannot be deleted, e.g., in 1 min, you might want to[ force-delete all pods](https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#delete_namespaced_pod) of the deployment with grace_period_seconds as 0.